### PR TITLE
Fixed drag and drop on browser Edge (bsc#1087181)

### PR DESCRIPTION
**Pull request by ghost**
This checkin includes:
1. added a check to see if it is MS Edge. Since we don't know if we will support IE, to make it simple,  only check edge for now.
2. When browser is Edge , transfer data format will be 'text', otherwise will
leave the way it is.
3. Also fixed a defect on role summary page when save the change of a
server with either auto discovered servers empty or manual added servers
empty

### CODE REVIEW

**Review by GarySmith — CHANGES_REQUESTED** [review]

File: src/utils/constants.js [hunk 1]
```diff
@@ -37,3 +37,5 @@ export const MODEL_SERVER_PROPS =  [
 ];
 
 export const MODEL_SERVER_PROPS_ALL = MODEL_SERVER_PROPS.concat(['role', 'uid']);
+
+export const IS_MS_EDGE = (/Edge/.test(navigator.userAgent)) ? true : false;
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/utils/constants.js]: ok, will simplify

File: src/pages/AssignServerRoles.js [hunk 2]
```diff
@@ -873,7 +875,7 @@ class AssignServerRoles extends BaseWizardPage {
         this.setState((prevState) => {
           return {serversAddedManually: prevState.serversAddedManually.concat([server])};
         });
-
+data
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/pages/AssignServerRoles.js]: oops...so leftover...odd it doesn't cause any trouble. will fix

File: src/pages/ServerRoleSummary.js [hunk 3]
```diff
@@ -82,13 +82,15 @@ class ServerRoleSummary extends BaseWizardPage {
...
     for (let list of ['autoServers', 'manualServers']) {
-      let idx = this.state[list].findIndex(s => server.uid === s.uid);
-      if (idx >= 0) {
-        let updated_server;
-        let tempList = this.state[list].slice();
-        updated_server = getMergedServer(tempList[idx], server, MODEL_SERVER_PROPS);
-        putJson('/api/v1/server', JSON.stringify(updated_server));
-        break;
+      if(this.state[list]) {
+        let idx = this.state[list].findIndex(s => server.uid === s.uid);
```
[the remark this answers is not in the document]
Comment (guglo) [reply src/pages/ServerRoleSummary.js]: ok will simplify

**Review by GarySmith — APPROVED** [review]

**Comment by GarySmith** [comment]
Waiting to merge this since you said you will also be adding IE support with polyfills

**Pull request merged**